charts/grafana/templates/_pod.tpl

```diff
@@ -1196,6 +1196,18 @@ containers:
         value: {{ (get .Values "grafana.ini").paths.plugins }}
       - name: GF_PATHS_PROVISIONING
         value: {{ (get .Values "grafana.ini").paths.provisioning }}
+      {{- if (.Values.resources.limits).cpu }}
+      - name: GOMAXPROCS
+        valueFrom:
+          resourceFieldRef:
+            resource: limits.cpu
+      {{- end }}
```

Comment:
With looking forward on Go 1.25 and https://go.dev/blog/container-aware-gomaxprocs this automatic discover should be favorite.

It additionally adjust the GOMAXPROCS, if the cgroups limits has been changed.

```suggestion
```